app/models/member.rb

```diff
@@ -86,4 +88,10 @@ def token_payload
       authentication_id: authentication.try(:id)
     }
   end
+
+  private
+
+  def existing_memberships_and_email_changed?
+    Member.where(email: email).where.not(id: id).any? && email_changed?
```

Comment:
I'd considered the idea of removing duplicates. To me, it's not such a trivial process, especially because of the number records we have. I'll sketch what a script could look like and we could run something in the background. I suspect it'd take a few hours to run.